sickbeard/subtitles.py

```diff
@@ -347,138 +645,58 @@ def __init__(self):
-                                               (os.path.join(root, filename), subtitle.provider_name,
-                                                subtitle.language.opensubtitles), logger.INFO)
-                                    save_subtitles(video, found_subtitles, directory=root, single=not sickbeard.SUBTITLES_MULTI)
-
-                                    subtitles_multi = not sickbeard.SUBTITLES_MULTI
-                                    subtitle_path = get_subtitle_path(video.name, None if subtitles_multi else subtitle.language)
-                                    if root:
-                                        subtitle_path = os.path.join(root, os.path.split(subtitle_path)[1])
-                                    sickbeard.helpers.chmodAsParent(subtitle_path)
-                                    sickbeard.helpers.fixSetGroupID(subtitle_path)
-
-                                    downloaded_languages.add(subtitle.language.opensubtitles)
-
-                                # Don't run post processor unless at least one file has all of the needed subtitles
-                                if not needs_subtitles(downloaded_languages):
-                                    run_post_process = True
-                            except Exception as error:
-                                logger.log(u'Error occurred when downloading subtitles for: {}. Error: {}'.format
-                                           (os.path.join(root, filename), ex(error)))
-                        elif subtitles_enabled is False:
-                            logger.log(u"Subtitle disabled for show: {}".format(filename), logger.DEBUG)
-
-            if run_post_process:
-                logger.log(u'Starting post-process with default settings now that we found subtitles')
-                processTV.processDir(sickbeard.TV_DOWNLOAD_DIR)
-        else:
-            logger.log(u'You must set a valid post-process folder in "Post Processing" settings', logger.WARNING)
+        for root, _, files in os.walk(sickbeard.TV_DOWNLOAD_DIR, topdown=False):
+            for filename in sorted(files):
+                filename = clear_non_release_groups(filename)
+
+                # Delete unwanted subtitles before downloading new ones
+                delete_unwanted_subtitles(root, filename)
+
+                if not isMediaFile(filename):
+                    continue
+
+                if processTV.subtitles_enabled(filename) is False:
+                    logger.log(u"Subtitle disabled for show: {0}".format(filename), logger.DEBUG)
+                    continue
+
+                video_path = os.path.join(root, filename)
+                release_name = os.path.splitext(filename)[0]
+                found_subtitles = download_best_subs(video_path, root, release_name, languages, subtitles=False,
+                                                     embedded_subtitles=False, provider_pool=pool)
```

Comment:
@ratoaq2 shouldn't be  `embedded_subtitles=bool(not sickbeard.EMBEDDED_SUBTITLES_ALL)`

so if video file has embedded subs and user don't want to ignore it, it can be PP with that sub only

as @NicoLeOca requested: can we get a checkbox "consider embedded subs as of default language"